src/Battlescape/TileEngine.cpp

```diff
@@ -4107,6 +4107,29 @@ bool TileEngine::medikitUse(BattleAction *action, BattleUnit *target, BattleMedi
 	return canContinueHealing;
 }
 
+/**
+ * Tries to conceal a unit. Only works if no unit of another faction is watching.
+ */
+bool TileEngine::tryConcealUnit(BattleUnit* unit)
+{
+	auto watchingUnits = getSpottingUnits(unit);
```

Comment:
This is probably wrong function to do it, it doing lot of calculations and even altering state of game, another thing this function return different values based on number of TU (as it generate list of "reactionshooters").

What exactly behavior you want this function to have?

Reply:
It should list all units (of the enemy faction) which are able to see this unit.
I can introduce a smaller function which does not alter state myself, if that is preferred.

Reply:
Looking on `calculateUnitsInFOV` I see that `removeFromVisibleUnits` is called when unit leave visible arena, this mean if not unit see this unit then all this lists should be empty already.